extensions/eventtap/internal.m

```diff
@@ -189,33 +191,45 @@ static int eventtap_start(lua_State* L) {
 ///  * None
 ///
 /// Returns:
-///  * None
+///  * The event tap object
 static int eventtap_stop(lua_State* L) {
     eventtap_t* e = luaL_checkudata(L, 1, USERDATA_TAG);
 
-    if (!e->running)
-        return 0;
-
-    remove_event(L, e->self);
-    e->self = LUA_NOREF;
-    e->running = false;
-
-    CGEventTapEnable(e->tap, false);
-    CFMachPortInvalidate(e->tap);
-    CFRunLoopRemoveSource(CFRunLoopGetMain(), e->runloopsrc, kCFRunLoopCommonModes);
-    CFRelease(e->runloopsrc);
-    CFRelease(e->tap);
+    if (e->tap && CGEventTapIsEnabled(e->tap)) {
```

Comment:
I presume you mean other than when the system disables the tap and issues a disabledByUserInput or disabledByTimeout event (because in this case, e->tap does exist and that's why the CGEventTapIsEnabled is called to check its status out... we don't want to _gc the object then because likely it will be restarted by the program (see #317)...

To my knowledge no, but perhaps it should be explicitly set to null in hs.evettap.new(...) like it is in hs.eventtap.stop().

Stop nulls it, and _gc erases it, and Accessibility being disabled sets it to NULL, so... the only assumption that is being made, I think, is when new() is called. Unless I'm missing your meaning.

I can resubmit later tonight.